docs/source/data/index.md

```diff
@@ -46,3 +46,25 @@ This section on handing data with Kedro concludes with an advanced use case, ill
 
 how_to_create_a_custom_dataset
 ```
+
+## `KedroDataCatalog` (experimental feature)
+
+As of Kedro 0.19.9, you can explore a new experimental feature — the `KedroDataCatalog`, an enhanced alternative to `DataCatalog`.
+
+At present, `KedroDataCatalog` replicates the functionality of `DataCatalog` and is fully compatible with the Kedro `run` command. It introduces several API improvements:
+* Simplified dataset access: `_FrozenDatasets` has been replaced with public `get` method to retrieve datasets.
+* Added dict-like interface: You can now use a dictionary-like syntax to retrieve, set, and iterate over datasets.
+
+For more details and examples of how to use `KedroDataCatalog`, see the Kedro Data Catalog page. The [documentation](./data_catalog.md) for `DataCatalog` remains relevant as `KedroDataCatalog` retains its core functionality with some enhancements.
+
+```{toctree}
+:maxdepth: 1
+
+kedro_data_catalog
+```
```

Comment:
I'd probably link this page in the sentence above: "For more details and examples of how to use `KedroDataCatalog`, see the Kedro Data Catalog page."

rather than add the toc tree here if possible.